packages/makeswift/src/integrate-next-app.ts

```diff
@@ -232,24 +244,80 @@ export default function Page({ snapshot }: Props) {
 
   // custom document
   if (
-    fs.existsSync(path.join(pagesFolder, `_document.${extension}`)) ||
-    fs.existsSync(path.join(pagesFolder, `_document.${extension}x`))
+    fs.existsSync(path.join(pagesFolder.absolute, `_document.${extension}`)) ||
+    fs.existsSync(path.join(pagesFolder.absolute, `_document.${extension}x`))
   ) {
     throw new MakeswiftError(
       'A custom document already exists, you will have to manually integrate: https://www.makeswift.com/docs/guides/manual-setup#set-up-custom-document',
     )
   }
   const customDocument = `export { Document as default } from '@makeswift/runtime/next'`
-  fs.writeFileSync(path.join(temporaryPagesFolder, `_document.${extension}`), customDocument)
+  fs.writeFileSync(path.join(pagesFolder.temporary, `_document.${extension}`), customDocument)
+
+  // register components file
+  fs.mkdirSync(path.join(temporaryDir, 'lib', 'makeswift'), { recursive: true })
+  const registerComponentsFile = `import { Style } from '@makeswift/runtime/controls'
+import { ReactRuntime } from '@makeswift/runtime/react'
+import HelloWorld from '../../components/hello-world'
+
+// Register your components here!
+
+ReactRuntime.registerComponent(HelloWorld, {
+  type: 'hello-world',
+  label: 'Hello, world!',
+  props: {
+    className: Style({ properties: Style.All }),
+  },
+})
+`
+  fs.writeFileSync(
+    path.join(
+      temporaryDir,
+      'lib',
+      'makeswift',
+      `register-components.${isTypeScript ? 'ts' : 'js'}x`,
+    ),
+    registerComponentsFile,
+  )
+
+  // hello world component
+  fs.mkdirSync(path.join(temporaryDir, 'components'))
+  if (isTypeScript) {
+    fs.writeFileSync(
+      path.join(temporaryDir, 'components', 'hello-world.tsx'),
+      `import React from 'react'
+
+export default function HelloWorld(props: { className?: string }) {
+  return <p {...props}>Hello, world!</p>
+}
```

Comment:
```suggestion
type Props = {
  className?: string
}

export function HelloWorld({ className }: Props) {
  return <p className={className}>Hello, world!</p>
}
```

Reply:
Note that the component is a named export so we'll need to update the `lib/makeswift/register-component.ts` file